src/lib.rs

```diff
@@ -605,35 +608,77 @@ pub fn build_with_params(parameters: CompileParameters) -> Result<(), Diagnostic
             output: config.to_owned(),
         });
 
-    let compile_options = CompileOptions {
+    let mut compile_options = CompileOptions {
```

Comment:
Comment above  is mainly to avoid this :)
No need to mutate the options, they should be fixed and derived early or passed as params